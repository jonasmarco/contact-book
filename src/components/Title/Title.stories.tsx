import { Meta, StoryFn } from '@storybook/react'

import Title, { Props } from '.'

export default {
  title: 'Title',
  component: Title,
  args: {
    children: 'Simple title'
  },
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta<typeof Title>

const Template: StoryFn<typeof Title> = (args: Props) => <Title {...args} />

export const SimpleTitle = Template.bind({})
SimpleTitle.args = {
  children: 'Simple title'
}
