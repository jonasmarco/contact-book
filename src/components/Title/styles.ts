import styled, { css } from 'styled-components'
import { customMedia } from '../MediaMatch'

export const Wrapper = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    font-size: ${theme.fonts.sizes.normal};
    font-weight: ${theme.fonts.weights.slim};
    letter-spacing: ${theme.text.letterSpacing};
    ${customMedia.greaterThan('tabletUp')`
      font-size: ${theme.fonts.sizes.extraLarge};
    `};
    ${customMedia.greaterThan('notebookUp')`
      font-size: ${theme.fonts.sizes.bigger};
    `};
  `}
`
