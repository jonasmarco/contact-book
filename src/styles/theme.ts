export default {
  grid: {
    container: '120rem',
    defaultPadding: '1rem',
    gutter: '1.5rem'
  },
  border: {
    radius: '1rem',
    rounded: '5rem'
  },
  fonts: {
    sizes: {
      small: '1.2rem',
      normal: '1.8rem',
      large: '2.2rem',
      extraLarge: '2.8rem',
      bigger: '4.2rem'
    },
    weights: {
      slim: 300,
      normal: 400,
      bold: 600,
      extraBold: 700
    }
  },
  colors: {
    black: '#02020A',
    white: '#FCFCFC',
    red: '#FF6565',
    blue: '#4FB6FF',
    darkBlue: '#273549',
    gray: '#596E7D',
    yellow: '#F5A623',
    green: '#33CC33',

    primary: '#4FB6FF',
    secondary: '#596E7D',
    tertiary: '#30425C'
  },
  layout: {
    mobile: {
      gap: '2rem'
    },
    tablet: {
      gap: '6rem'
    },
    notebook: {
      gap: '10rem'
    },
    gapForm: '1rem'
  },
  text: {
    letterSpacing: '0.135rem',
    lineHeight: '2.8rem'
  },
  margin: {
    normal: '1.8rem',
    textMargin: '1.2rem'
  },
  layers: {
    base: 0,
    behindMenu: 5,
    menu: 10,
    modal: 100
  }
} as const
