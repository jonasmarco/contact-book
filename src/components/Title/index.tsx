import * as S from './styles'

export type Props = {
  children: string
}

const Title = ({ children }: Props) => {
  return <S.Wrapper>{children}</S.Wrapper>
}

export default Title
