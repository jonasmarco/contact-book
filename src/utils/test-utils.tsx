import { ReactElement } from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'

import { ThemeProvider } from 'styled-components'
import theme from '../styles/theme'

const customRender = (ui: ReactElement) =>
  render(
    <Router>
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </Router>
  )

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
