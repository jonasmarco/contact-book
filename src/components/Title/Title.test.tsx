import { render } from '../../utils/test-utils'

import theme from '@/styles/theme'
import Title from '.'

describe('Title Component', () => {
  // (renderiza sem erros):
  // ensures that the Title is rendered without errors
  test('is rendered without error', () => {
    render(<Title>My Title</Title>)
  })

  // (renderiza com o texto correto):
  // ensures that the Title is rendered with the correct text
  test('is rendered the correct text', () => {
    const { getByText } = render(<Title>My Title</Title>)
    const titleElement = getByText('My Title')
    expect(titleElement).toBeInTheDocument()
  })

  // (renderiza com o estilo correto):
  // ensures that the Title is rendered with the correct styles
  test('applies the correct styles', () => {
    const { getByText } = render(<Title>My Title</Title>)
    const titleElement = getByText('My Title')
    expect(titleElement).toHaveStyle(`
      color: ${theme.colors.primary};
      font-size: ${theme.fonts.sizes.normal};
      font-weight: ${theme.fonts.weights.slim};
      letter-spacing: ${theme.text.letterSpacing};
    `)
  })

  // (renderiza com o estilo correto em diferentes resoluções):
  // ensure that the Title is rendered with the correct styles according to the resolution
  test('applies different styles for tablet and notebook screens', () => {
    const { getByText } = render(<Title>My Title</Title>)
    const titleElement = getByText('My Title')
    expect(titleElement).toHaveStyleRule(
      'font-size',
      theme.fonts.sizes.extraLarge,
      {
        media: '(min-width: 769px)'
      }
    )
    expect(titleElement).toHaveStyleRule(
      'font-size',
      theme.fonts.sizes.bigger,
      {
        media: '(min-width: 1367px)'
      }
    )
  })

  // (snapshot)
  test('matches the snapshot', () => {
    const { asFragment } = render(<Title>My Title</Title>)
    expect(asFragment()).toMatchSnapshot()
  })
})
