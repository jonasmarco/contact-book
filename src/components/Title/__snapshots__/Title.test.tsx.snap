// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #4FB6FF;
  font-size: 1.8rem;
  font-weight: 300;
  -webkit-letter-spacing: 0.135rem;
  -moz-letter-spacing: 0.135rem;
  -ms-letter-spacing: 0.135rem;
  letter-spacing: 0.135rem;
}

@media (min-width:769px) {
  .c0 {
    font-size: 2.8rem;
  }
}

@media (min-width:1367px) {
  .c0 {
    font-size: 4.2rem;
  }
}

<h1
    class="c0"
  >
    My Title
  </h1>
</DocumentFragment>
`;
